modules/end-user-guide/pages/adding-a-vscode-extension.adoc

```diff
@@ -0,0 +1,167 @@
+
+. Specify the memory limit for the extension.
++
+[source,yaml,subs="+quotes"]
+----
+
+  - id: redhat/java  
+    override:  
+      sidecar:  
+        memoryLimit: 2Gi
+----
+. {prod-short} will automatically install the Red Hat Visual Studio Code Java extension in the {prod-short} workspace and increase the memory of the workspace by two gigabytes.
+
+=== Override default preferences
+
+Override the default preferences of the Visual Studio Code extension for the workspace.
+
+.Procedure
+
+. Add `override` section in the `.che/che-theia-plugins.yaml` file under the `id` of the extension.
+
+. Specify the preferences in the `Preferences` section.
++
+[source,yaml,subs="+quotes"]
+----
+  - id: redhat/java  
+    override:  
+      preferences:
+        java.server.launchMode: LightWeight
+----
+. {prod-short} will automatically install Red Hat Visual Studio Code Java extension in the {prod-short} workspace and sets the `java.server.launchMode` preference to `LightWeight`.
+
+This method allows to override the default preferences or memory limit of the Visual Studio Code extension for the workspace.
+
+Note: It is also possible to define the preferences in the `.vscode/settings.json` file.
+
+[id="visual-studio-code-extensions-in-devfile"]
+== Defines Visual Studio Code extensions in the devfile
+
+The devfile v2 specification is not allowing to define the Visual Studio Code extensions in the devfile. But it is possible to define some attributes in the devfile.
+
+This definition is helpful when it is not possible to add extra files in the repository but is not the recommended way.
+
+Defines the Visual Studio Code extensions by inlining them in the devfile.
+
+=== Inlining `.vscode/extensions.json` file
+
+.Procedure
+
+. Edit the `devfile.yaml` file and add if it is missing the `attributes` section at the root of the devfile.
+
+. Add `.vscode/extensions.json` and appends a | character after the colon separator
+
+. Paste the content of the `.vscode/extensions.json` file
+
+Example
+
+[source,yaml,subs="+quotes"]
+----
+schemaVersion: 2.2.0  
+metadata:  
+  name: my-example  
+attributes:  
+  .vscode/extensions.json: |  
+    {  
+      "recommendations": [  
+        "redhat.java"  
+      ]  
+    }
+----
+
+=== Inlining `.che/che-theia-plugins.yaml` file
+
+. Edit `devfile.yaml` file and add if missing the `attributes` section at the root of the devfile
+
+. Add `.vscode/extensions.json` and appends a | character after the colon separator
+
+. Paste the content of the `.che/che-theia-plugins.yaml` file
+
+Example:
+
+[source,yaml,subs="+quotes"]
+----
+schemaVersion: 2.2.0  
+metadata:  
+  name: my-example  
+attributes:  
+  .che/che-theia-plugins.yaml: |  
+    - id: redhat/java
+----
+
+[id="plug-in-registries"]
+== Plug-in registries
+
+{prod-short} provides a default plug-in registry installed on every {prod-short} instance.
+
+When using `Che-Theia` as IDE, it will search and install the extensions from there.
```

Comment:
my understanding that `che-theia` does not support the OpenVSX registry, right?

Reply:
correct